### PR TITLE
add precompiled headers

### PR DESCRIPTION
**Pull request by AndreyMlashkin**

### CODE REVIEW

File: crypto3/.gitignore [hunk 1]
```diff
@@ -250,7 +250,6 @@ TSWLatexianTemp*
 
 # Precompiled Headers
 *.gch
-*.pch
```
Comment (akokoshn) [inline crypto3/.gitignore:253]: Let's make single common .gitignore in the root of placeholder

File: crypto3/libs/algebra/include/nil/crypto3/algebra/algebra.pch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#include <nil/crypto3/algebra/curves/pallas.hpp>
```
Comment (akokoshn) [inline crypto3/libs/algebra/include/nil/crypto3/algebra/algebra.pch:1]: auto generated `*.pch `should be ignored, right?

File: crypto3/.gitignore (hunk 1 above)
Comment (AndreyMlashkin) [reply crypto3/.gitignore:253]: ok, I will do a general cmake-cleanup together with export statements unification

File: crypto3/libs/algebra/include/nil/crypto3/algebra/algebra.pch (hunk 2 above)
Comment (AndreyMlashkin) [reply crypto3/libs/algebra/include/nil/crypto3/algebra/algebra.pch:1]: This whole PR is yet highly experimental

**Pull request closed without merging**